pkg/tar/tar_test.go

```diff
@@ -692,3 +721,84 @@ func checkTime(path string, time time.Time) error {
 	}
 	return nil
 }
+
+func TestExtractTarHardLink(t *testing.T) {
+	for _, insecure := range []bool{false, true} {
+		if !insecure && !sys.HasChrootCapability() {
+			t.Skipf("chroot capability not available. Disabling test.")
```

Comment:
This isn't quite right because you're skipping _both_ test cases (insecure && !insecure)